src/test/java/it/gov/pagopa/initiative/statistics/events/consumers/MerchantCountersTransactionMessagesListenerTest.java

```diff
@@ -0,0 +1,163 @@
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+
+class MerchantCountersTransactionMessagesListenerTest extends BaseMerchantStatisticsMessageListenerTest {
+
+    @SpyBean
+    private MerchantTransactionStatisticsService merchantTransactionStatisticsService;
+
+    @Test
+    @Override
+    void test(){
+        super.test();
+    }
+
+    @Override
+    protected StatisticsEvaluationService getStatisticsEvaluationServiceSpy() {
+        return merchantTransactionStatisticsService;
+    }
+
+    @Override
+    protected String getStatisticsMessagesTopic() {
+        return topicMerchantCountersTransaction;
+    }
+
+    @Override
+    protected String getStatisticsMessagesGroupId() {
+        return groupIdMerchantCountersTransaction;
+    }
+
+    @Override
+    protected Function<MerchantInitiativeCounters, Long> getGetterCounter() {
+        return MerchantInitiativeCounters::getTotalProvidedCents;
+    }
+
+    @Override
+    protected BiConsumer<MerchantInitiativeCounters, Long> getSetterCounter() {
+        return MerchantInitiativeCounters::setTotalProvidedCents;
+    }
+
+    @Override
+    protected Function<MerchantInitiativeCounters, List<CommittedOffset>> getGetterStatisticsCommittedOffsets() {
+        return MerchantInitiativeCounters::getTrxCommittedOffsets;
+    }
+
+    @Override
+    protected BiConsumer<MerchantInitiativeCounters, List<CommittedOffset>> getSetterStatisticsCommittedOffsets() {
+        return MerchantInitiativeCounters::setTrxCommittedOffsets;
+    }
+
+    @Override
+    protected List<TransactionEvaluationDTO> buildValidEntities(int bias, int size, String initiativeId) {
+        List<TransactionEvaluationDTO> out = new ArrayList<>(buildValidTransactionEvaluationEntities(bias, size, initiativeId));
+        out.forEach(t -> {
+            t.setRewards(new HashMap<>(t.getRewards()));
+            t.getRewards().put(initiativeId+"_2", new Reward(initiativeId+"_2", "ORGANIZATIONID_"+initiativeId, BigDecimal.valueOf(2)));
+        });
+
+        out.add(buildValidTransactionEvaluationEntity(bias, initiativeId, "MERCHANTID2"));
+        out.add(buildValidTransactionEvaluationEntity(bias, initiativeId, null));
+
+        return out;
+    }
+
+    private TransactionEvaluationDTO buildValidTransactionEvaluationEntity(int bias, String initiativeId, String merchantId) {
+        TransactionEvaluationDTO out = super.buildValidTransactionEvaluationEntity(bias, initiativeId);
+        out.setMerchantId(merchantId);
+        return out;
+    }
+
+    @Override
+    protected List<TransactionEvaluationDTO> buildSkippedEntities(int bias, int size) {
+        List<TransactionEvaluationDTO> out = TransactionEvaluationMessagesListenerTest.buildTransactionSkippedEntities(bias, size);
```

Comment:
could you try to retun a List where as first, you put some valid payload, having another merchantId (filled in order to be published into partition 1)

it's important that these new requests are put before the ASIS skipped, in order to have greater offsets for MERCHANTID skipped messages